arbitrum-docs/bold/bold-gentle-introduction.md

```diff
@@ -0,0 +1,139 @@
+3. **Multi-level, interactive dissection begins:** To resolve the dispute, the disagreeing entities will need to come to an agreement on what the *actual, correct* asserted state should be. [It would be tremendously expensive to re-execute](https://docs.arbitrum.io/inside-arbitrum-nitro/#why-interactive-proving-is-better) and compare everything from `Block A → Block Z` and `Block A → Block Y`, especially since there could be potentially millions of transactions in between `A`, `Z`, and `Y`. Instead, entities take turns bisecting their respective *history commitments* until they arrive at a single step of instruction where an arbiter, like Ethereum, can declare a winner. Note that [this system is very similar to how challenges are resolved on Arbitrum chains today](https://docs.arbitrum.io/inside-arbitrum-nitro/#challenges) - BOLD only changes some minor, but important, details in the resolution process. Let’s dive into what happens next:
+    1. **Block challenges**: when a challenge is opened, the edge is called a *level-zero edge* since it is at the granularity of Arbitrum blocks. The disputing parties take turns bisecting their history commitments until they identify the specific block that they disagree on. 
+    2. **Big-step challenges:** now that the parties have narrowed down their dispute to a single block, that we call `Block B`, the back-and-forth bisection exercise continues within that block. Note that `Block B` is claimed by all parties to be some state after the initial state `Block A` but before the final states `Block Z` and `Block Y`. This time however, the parties will narrow down on a specific *range* of instructions for the state transition function within the block - essentially working towards identifying a set of instructions that their disagreement lies within. This range is currently defined to be 2^20 steps of WASM instructions, which is the assembly of choice for validating Arbitrum chains.
+    3. **One-step challenge:** within that range of 2^20 instructions, the back and forth bisecting continues until all parties arrive at a single step of instruction that they disagree on. At this point in time, parties agree on the initial state of Arbitrum before the step but disagree on the end state 1 step immediately after. Remember that since Arbitrum’s state is entirely deterministic, there is only 1 correct end state.
+4. **One-step proof:** Once a challenge is isolated to a dispute about a single step, the honest party's edge can be proven correct with a *one-step proof*. A one-step proof is, in theory, a proof that a single step of computation results in a particular state. The honest party will submit such a proof, whose correctness is easily and quickly validated via execution on Ethereum. The honest edge will thus be confirmed, and the dishonest one rejected.
```

Comment:
```suggestion
4. **One-step proof:** Once a challenge is isolated to a dispute about a single step, the honest party's edge can be proven correct with a *one-step proof*. A one-step proof is a proof that a single step of computation results in a particular state. The honest party will submit such a proof, whose correctness is easily and quickly validated via execution on Ethereum. The honest edge will thus be confirmed, and the dishonest one rejected.
```

And also, the disputed single step is actually executed on Ethereum, rather than submitted by the honest party. I.e., both parties have submitted what they think is the resulting state after executing that step, and that step is executed by the OneStepProof smart contract on Ethereum (or the parent chain), to be completely unbiased/objective.

Reply:
makes sense, how does this sound?

> Once a challenge is isolated down to a dispute about a single step, both parties run that step to produce, and then submit, a *one-step proof* to the `OneStepProof` smart contract on the parent chain (e.g. Ethereum). A one-step proof is a proof that a single step of computation results in a particular state. The smart contract on the parent chain will execute the disputed step to validate the correctness of a submitted proof from the two parties. It is at this point that the honest party's proof will be deemed valid and it's tree of edges will get confirmed, while the dishonest party has their edges rejected.

Reply:
Sounds much better, but I think we should have someone else's take on it too, since I'm not sure about the "proof submission" part. I have limited knowledge about this, but in my mind, only initialState and instructionToExecute are sent to the smart contract, and the sc returns the actual state that results from that operation, which will make the honest party win the challenge.